storey/table.py

```diff
@@ -244,6 +275,148 @@ def __getitem__(self, key):
          """
         return self._get_static_attrs(key)
 
+    async def _flush_worker(self):
+        try:
+            while not self._terminated:
+                await asyncio.sleep(self._flush_interval_secs)
+                for key in self._changed_keys:
+                    await self._persist(_PersistJob(key, None, None))
+        except BaseException as ex:
+            self._flush_exception = ex
+
+    async def _persist_worker(self):
+        task = None
+        received_job_count = 0
+        self_sent_jobs = {}
+        try:
+            while True:
+                jobs = self_sent_jobs.pop(received_job_count, None)
+                if jobs:
+                    task = jobs[0]
+                    if len(jobs) > 1:
+                        self_sent_jobs[received_job_count] = jobs[1:]
+                else:
+                    task = await self._q.get()
+                    received_job_count += 1
+                    if task is _termination_obj:
+                        if received_job_count in self_sent_jobs:
+                            await self._q.put(_termination_obj)
+                            continue
+                        for key, pending_event in self._pending_by_key.items():
+                            if pending_event.pending and not pending_event.in_flight:
+                                for job in pending_event.pending:
+                                    resp = await self._internal_persist_key(key, job.data)
+                                    self._changed_keys.discard(key)
```

Comment:
Can't it be that the key was changed again while `await self._internal_persist_key` suspended execution?

Reply:
not really, this flow will execute when a termination_obj is received so no updates will come after it